### PR TITLE
✨ add attribute row

### PR DESCRIPTION
**Pull request by iantrich**
## Proposed change

Add a row to display entity attributes. Currently requires creating a template sensor which a big ask for new users.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
entities:
  - type: attribute
    entity: light.bed_light
    attribute: brightness
    default: '0'
    prefix: 'Brightness: '
    suffix: '%'
    tap_action:
      action: toggle
type: entities
```
Peek 2020-02-01 09-26

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/special-rows/hui-attribute-row.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+class HuiAttributeRow extends LitElement implements LovelaceRow {
+  @property() public hass?: HomeAssistant;
+  @property() private _config?: AttributeRowConfig;
+
+  public setConfig(config: AttributeRowConfig): void {
+    if (!config) {
+      throw new Error("Configuration error");
+    }
+    if (!config.entity) {
+      throw new Error("entity not defined");
```
Comment (zsarnett) [inline src/panels/lovelace/special-rows/hui-attribute-row.ts]: Picky: Capitalize `entity`

File: src/panels/lovelace/special-rows/hui-attribute-row.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    }
+    if (!config.attribute) {
+      throw new Error("attribute not defined");
```
Comment (zsarnett) [inline src/panels/lovelace/special-rows/hui-attribute-row.ts]: Picky: Capitalize `attribute`

File: src/panels/lovelace/entity-rows/types.ts [hunk 3]
```diff
@@ -59,3 +60,9 @@ export interface ConditionalRowConfig extends EntityConfig {
   row: EntityConfig;
   conditions: Condition[];
 }
+export interface AttributeRowConfig extends EntityConfig {
+  attribute: string;
+  prefix?: string;
```
Comment (zsarnett) [inline src/panels/lovelace/entity-rows/types.ts]: Should we align this to be more like the Gauge Card Configuration? `Unit` and `name`? Maybe `attribute_name`

Also, is the default necessary for some attributes?

Comment (iantrich) [reply src/panels/lovelace/entity-rows/types.ts]: I prefer what is currently set

**Comment by iantrich** [comment]
anything else needed on this to get it merged?

File: src/panels/lovelace/special-rows/hui-attribute-row.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+            this._config.attribute
+          )}</hui-warning
+        >
+      `;
+    }
+
+    return html`
+      <hui-generic-entity-row .hass=${this.hass} .config=${this._config}>
+        <div>
+          ${this._config.prefix} ${attribute || this._config.default}
```
Comment (bramkragten) [inline src/panels/lovelace/special-rows/hui-attribute-row.ts]: What is the use-case of default?

Comment (iantrich) [reply src/panels/lovelace/special-rows/hui-attribute-row.ts]: default is for attributes that are not always set, e.g. brightness on a light

Comment (bramkragten) [reply src/panels/lovelace/special-rows/hui-attribute-row.ts]: I think we should just not show it in that case instead of a default config option

Comment (iantrich) [reply src/panels/lovelace/special-rows/hui-attribute-row.ts]: That's a pretty useless row at that point then and confusing as well, I don't like the idea of that.

Comment (zsarnett) [reply src/panels/lovelace/special-rows/hui-attribute-row.ts]: In the entity card we are displaying it as Unknown. We should align them.

Comment (bramkragten) [reply src/panels/lovelace/special-rows/hui-attribute-row.ts]: Or just a `-` would also do I think?

File: src/panels/lovelace/special-rows/hui-attribute-row.ts [hunk 5]
```diff
@@ -58,24 +58,17 @@ class HuiAttributeRow extends LitElement implements LovelaceRow {
...
     return html`
       <hui-generic-entity-row .hass=${this.hass} .config=${this._config}>
-        <div>
-          ${this._config.prefix} ${attribute || this._config.default}
-          ${this._config.suffix}
-        </div>
+        ${attribute
+          ? html`
+              <div>
+                ${this._config.prefix} ${attribute} ${this._config.suffix}
```
Comment (bramkragten) [inline src/panels/lovelace/special-rows/hui-attribute-row.ts]:
```suggestion
                ${this._config.prefix} ${attribute || "-"} ${this._config.suffix}
```

File: src/translations/en.json [hunk 6]
```diff
@@ -2127,7 +2127,8 @@
         },
         "warning": {
           "entity_not_found": "Entity not available: {entity}",
-          "entity_non_numeric": "Entity is non-numeric: {entity}"
+          "entity_non_numeric": "Entity is non-numeric: {entity}",
+          "attribute_not_found": "Attribute not available: {attribute}"
```
Comment (bramkragten) [inline src/translations/en.json:2131]: Not used

**Review by bramkragten — APPROVED** [review]

**Comment by frenck** [comment]
@iantrich Can you make sure we get docs for this? 👍

**Comment by iantrich** [comment]
@frenck done

**Pull request merged**